toolchain/check/context.cpp

```diff
@@ -368,9 +473,24 @@ auto Context::LookupQualifiedName(Parse::NodeId node_id, SemIR::NameId name_id,
   }
 
   if (required && !result.inst_id.is_valid()) {
-    if (!has_error) {
+    if (has_error) {
+      return {.specific_id = SemIR::SpecificId::Invalid,
+              .inst_id = SemIR::InstId::BuiltinError};
+    }
+
+    if (prohibited_accesses.empty()) {
       DiagnoseNameNotFound(node_id, name_id);
+    } else if (access_info.has_value()) {
```

Comment:
Does this need to be checked, given prohibited_accesses is non-empty? i.e.:

```suggestion
    } else {
```